unlock-app/src/components/interface/checkout/main/checkoutHookUtils.ts

```diff
@@ -71,3 +73,73 @@ export const getOnPurchaseHookTypeFromPaywallConfig = (
     return 'promocode'
   }
 }
+
+const saveToLocalStorage = (data: any) => {
+  const hooks = localStorage.getItem('hooks')
+  const parsed = (hooks && JSON.parse(hooks)) ?? []
+  localStorage.setItem(
+    'hooks',
+    JSON.stringify([...parsed, { id: parsed.length, ...data }])
+  )
+}
+
+let prevBody: string | null = null
+export const postToWebhook = async (body: any, config: any, event: string) => {
+  const url = config?.hooks && config.hooks[event]
+
+  if (!url) return
+
+  if (JSON.stringify(body) === prevBody) {
+    return
+  }
+
+  prevBody = JSON.stringify(body)
+
+  const sendWebhookRequest = async (attempt: number) => {
+    await axios.post(url, body)
+    const text = `Sent ${event} event data to ${url} on attempt ${attempt}`
+    toast.success(text)
+
+    const data = { text, created: new Date().toISOString(), success: true }
+    saveToLocalStorage(data)
+  }
+
+  const retryRequest = async (maxRetries: number) => {
+    let lastError: any
+    let attempt = 0
+
+    while (attempt < maxRetries) {
+      try {
+        attempt++
+        await sendWebhookRequest(attempt)
+        return
+      } catch (error) {
+        lastError = error
+        toast.error(
+          `Could not post ${event} event data to ${url}. Attempt ${attempt}`
+        )
+        if (attempt < maxRetries) {
+          const delay = attempt === 1 ? 1000 : 3000
+          await new Promise((resolve) => setTimeout(resolve, delay))
+        }
+      }
+    }
+
+    setTimeout(() => {
+      if (lastError && attempt === maxRetries) {
+        const text = `Failed to post ${event} event data to ${url}`
+        toast.error(text)
+        const data = { text, created: new Date().toISOString(), success: false }
+        saveToLocalStorage(data)
+      }
+    }, 1000)
+
+    throw lastError
+  }
+
+  try {
+    await retryRequest(3)
+  } catch (error) {
+    console.error('Webhook request failed:', error)
+  }
+}
```

Comment:
This is a security issue as well as reliability. No calls should be made to user given urls from frontend. It should go through a proxy and have some sort of verification. 

Also exponential backoff is preferred in cases such as this and should account for failure by storing the request data in backend. This should be easy to add as a worker to graphile-worker and start from backend.